src/store/CategoricalDisplayStore.js

```diff
@@ -92,14 +92,14 @@ export default class CategoricalDisplayStore {
         // get first non-empty category
         let idx = i - 1;
         let id;
-        do {
+        while (idx >= 0) {
             id = this.filteredItems[idx].categoryId;
             if (this.categoryElements[id] !== null) {
                 break;
             }
             idx--;
-        } while (idx >= 0);
```

Comment:
It's been a while i've not seen a do while, we should have kept it :haircut_man: